### PR TITLE
[feat] : img augmentor logic : @augmentation.py

### PR DESCRIPTION
**Pull request by MoldyLunchBox**

### CODE REVIEW

File: 001-Augmentation.py [hunk 1]
```diff
@@ -1 +1,278 @@
...
+                operation(0.8)
+            elif operation.__name__ == 'flip_vertical':
+                operation()
+            elif operation.__name__ == 'adjust_contrast':
+                factor = random.uniform(0.5, 2.0)
+                operation(factor)
+
+
+import sys
+import argparse
```
Comment (aallali) [inline 001-Augmentation.py]: move the imports to head section of the file

File: 001-Augmentation.py [hunk 2]
```diff
@@ -1 +1,278 @@
...
+            for image in data['image_paths'][leaf]['images']:
+                imagePath = f"{path_to_folder}/{image}"
+                num_augmentations = augmentations_total if size - files_generated > augmentations_total else max(size - files_generated,0)
+                files_generated += num_augmentations
+                augment_and_save_single_image(imagePath, num_augmentations, export_path)
+            
+    else:
+        print(f"Error: {input_path} is not a valid file or directory.")
+        sys.exit(1)
+def main():
```
Comment (aallali) [inline 001-Augmentation.py]: add 2 lines between the functions

File: 001-Augmentation.py [hunk 3]
```diff
@@ -1 +1,278 @@
...
+    parser.add_argument("-size", type=int, default=augmentations_total, help="Number of augmentations to generate (default: 7 and max is number of images multiplied by 7)")
+    parser.add_argument("-export_location", type=str, default=None, help="images export location (default: image or directory path)")
+    args = parser.parse_args()
+    input_path, size, export_location = [args.input_path, args.size, args.export_location]
+    
+    ft_augmentation(input_path, size, export_location)
+    
+    
+
+
```
Comment (aallali) [inline 001-Augmentation.py]: too much spacing

File: 001-Augmentation.py [hunk 4]
```diff
@@ -1 +1,278 @@
...
+        
+        num_augmentations = augmentations_total if size - files_generated > augmentations_total else size - files_generated
+
+        # Apply some random augmentations
+        augmentor.some_augmentations(num_augmentations)
+        
+        # Save augmented images
+        augmentor.save_images()
+        files_generated += num_augmentations
+
```
Comment (aallali) [inline 001-Augmentation.py]: respect NORM in function, no additonal new lines, and make en empty line between variables declarations and rest of code

**Comment by aallali** [comment]
in order to solve the issue of the code norme, we will use `FLAKE8` as requested in the subject
you can install it via `pip install flake8` its a tool for style guide enforcement
search about it for more infos

**Comment by MoldyLunchBox** [comment]
ok that is indeed done

**Comment by aallali** [comment]
LGTM

**Pull request merged**